src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.userinvalidexeption.UserInvalidExeption;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+
+    @BeforeEach
+    void setUp() {
+        StorageDao storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_ValidUser_Ok() {
+        User validUser = new User();
+        validUser.setLogin("john_doe");
+        validUser.setPassword("strongPassword");
+        validUser.setAge(25);
+
+        User registeredUser = registrationService.register(validUser);
+
+        assertNotNull(registeredUser.getId());
+        assertEquals(validUser.getLogin(), registeredUser.getLogin());
+        assertEquals(validUser.getPassword(), registeredUser.getPassword());
+        assertEquals(validUser.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    void register_UserWithShortLogin_Not_Ok() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("short");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(30);
+
+        assertThrows(UserInvalidExeption.class, () ->
+                registrationService.register(userWithShortLogin));
+    }
+
+    @Test
+    void register_UserWithExistingLogin_Not_ok() {
+        User existingUser = new User();
+        existingUser.setLogin("existing_user");
+        existingUser.setPassword("validPassword");
+        existingUser.setAge(28);
+
+        registrationService.register(existingUser);
```

Comment:
we should use low-level methods of `storage` or `StorageDao` to arrange the existing user test data instead, because `register()` is actually the method under the test and we can not guarantee that it is working properly, check common mistakes for details

Reply:
removed